### ISSUE TITLE
add possibility to specify host ip address/info (id) manually

### ISSUE THREAD

**Issue by lmdc45** [Swamp-Ig/homeassistant_izone#35]
### Checklist

- [X] I have filled out the template to the best of my ability.
- [X] This only contains 1 feature request (if you have multiple feature requests, open one feature request for each feature request).
- [X] This issue is not a duplicate feature request of [previous feature requests](https://github.com/ludeeus/integration_blueprint/issues?q=is%3Aissue+label%3A%22Feature+Request%22+).

### Is your feature request related to a problem? Please describe.

I am always frustrated when I try to connect to the hub while home assistant is running from a docker container.

It is said:
Network settings
The iZone system uses UDP broadcasts over the local network to find and communicate with iZone devices. For this to work properly, UDP port 12107 must be able to be broadcasted on, 7005 needs to be listened to for broadcasted messages, and TCP port 80 for HTTP data to the bridge. The integration currently listens on 0.0.0.0 and broadcasts to all broadcast IPv4 local addresses, which is not configurable.

But I cant find a way to fetch the hub network on the host wifi interface, ha is only looking at the docker interface

### Describe the solution you'd like

add possibility to specify host ip address/info (id) manually

### Describe alternatives you've considered

trying to run the container with --net=host option and then change the network adapter but could not access ha UI

### Additional context

ha is run on docker

**Comment by lmdc45** [comment]
found a way might help anyone else with same issue:

on the host execute:
echo "IASD" | socat - UDP-DATAGRAM:192.168.1.255:12107,broadcast

verify you got some result that looks like ASPort....

then execute 
echo "IASD" | socat - UDP-DATAGRAM:192.168.1.255:12107,broadcast |  socat - UDP-DATAGRAM:172.17.0.2:7005

replace 192.168.1.255 by your multicast address, replace 172.17.0.2 by your docker container ip address. execute the command after clicking on the izone 'submit' button

that worked for me

**Comment by Swamp-Ig** [comment]
Heya.

This custom component isn't really in active development, I've never even released it. The native HA integration is essentially stalled too. I really would need someone to co-develop with me to get it going again. It's HA policy to always have code reviews for every commit and the lead devs are too busy to do it.

I think it might actually be better to move the whole thing out as a custom component, and distribute through HACS, that way it's not tied into the mainline HA release cycle.

If you'd be interested on working on this with me that would be great.

**Comment by lmdc45** [comment]
I am actually using the HA integration but google forwarded me here.
happy to try helping out however would probably not be helpful at all though as I am not into python at all, but can use chatgpt

**Comment by lmdc45** [comment]
Gday
Tried to install via HACS but could not, let me know if there is anything I need to do so I can give it a try

**Comment by Swamp-Ig** [comment]
Yeh I hadn't even gotten as far as to do that yet!

**Comment by lmdc45** [comment]
could you summarise quickly how this work ?
I would like to add a ha tool to set and monitor airflows. I find that current system is not incredible, as if only one room is left to heat or cool then airflow is very high even at say max 60%, which is not high at all and might be too slow when all rooms are open. as room can be close to indoor unit living higher value means too much air is going through that room first when the all house should get it evenly.
so would like to be able to set a tool that modify airflow based on an max l/s calculated (approximated) for each room, not only a max set % opening, potentially opening some fan only room as well to evacuate extra air.

If you could add a step by step guide on how I should proceed I would appreciate. never worked with ha components before.

I also realised my system is not giving correct damper position, just giving 50% for all of them. apparently it is a bug in version I have, not sure if can be fixed as izone support is quite slow but might be hard to do without that correct data
(System device list:
MAC3 435
CCTSL-5L6 (this) v2.20
C225-1C4 v3.56
CACUM v2.6 (G)
COCB-5H8 v4.18)

I also realised the wifi bridge is sometime really lagging before updating the data. sometimes 30-30mns wich make hard to automate things. I will try to find a way to get it connected with wire to the izone system and not only radio that might be one of the reason

**Comment by Swamp-Ig** [comment]
Hey, 

This isn't a working repo at the moment. The IZone integration is already in HA. There's no real way to set the zone flow with the addon, it could be made to work like that but I found working with HA a bit too hard since you have to get reviews and the main devs aren't that interested.

**Comment by lmdc45** [comment]
ok even not planning to make it on HACS?

I am using this componant already really handy. normal set up of izone heat up the home and then cut it all when all rooms reach set point, loosing a lot of heat in the roof leftover from the exchanger due to inertia of heating.
thanks to HA when all rooms reach set temp, I can set to fan only while supply temp is above 25degree. that still gives about 5 to 10 mns of heat otherwise lost outside or in the roof.
it is also helping keep the compressor in good condition as a 'soft landing' re loosing all heat through exchanger is better than loosing it outside through relief valve when system if abruptly cut.

Not sure why this is not a default option but glad ha could solve this.

**Comment by Swamp-Ig** [comment]
Yeh... Planning. Not actively working on though. I have many projects on the go :) 

If you wanted to work on the HA code together, then we can approve each other's PRs and get some progress going. 

Another option would be to just remove the integration from HA and install it through HASS, but to use HASS repo you can't have the integration in mainline HA, so it would have to be taken out. The downside to that is HA tends to change their API, and when they do that they update the integrations in it to work. As a HASS addon you don't get that.

I got interested in fixing up the HA code, but it got bogged down in review and never went through and I lost interest, what was there kinda did what I wanted anyhow.

**Comment by lmdc45** [comment]
Yes it is already pretty good and can do all the climate stuffs which is good enough.
However happy to try get the HA integration updated as much as possible don't as much time as well
Where should I start?

**Comment by Swamp-Ig** [comment]
If [PR #89730](https://github.com/home-assistant/core/pull/89730) was accepted that would be a good start.
There's also this one: [#89579]((https://github.com/home-assistant/core/pull/89579)

**Comment by Swamp-Ig** [comment]
I've got code that works with the power meter too, it's really waiting on that stuff.

**Comment by Swamp-Ig** [comment]
BTW: There's a non-linear relationship between airflow and damper position, the flow goes up rapidly as the damper gets opened, then flattens out until full flow. It's more sinusoidal than linear. The flow also depends on the damper position of all the other dampers in the system, the fan speed, and if the filter has been cleaned recently. That's probably why the system tends to use bang-bang control rather than smooth flow variation. To do it flow based you'd really need to put a flow sensor in the line and have a feedback loop.

**Comment by lmdc45** [comment]
Don't understand much re PR at first look...
Don't know anything about how HA works internally yet but will have a closer look and see how changes can be tested quickly.

Concerning the flow I would only use number of open rooms to get rough estimate. IE one room should open at max 40 if that's the only one open , 50 if there is others and 80 if all are open. Could prob just control fan speed but my install can't control it, just have a fan on/fan off. Looking at changing that though.

Issue is at night when only one or two bedroom is getting heat for instance it will make a big noise and go too quick and inertia will push the temp up too much. So main goal is to mitigate that.

Having said that I don't have any power monitor in my install and can't get damper position for now so testing will be challenging. Still no answer from izone support

**Comment by Swamp-Ig** [comment]
My fan speed is set to low, rather than auto. Since the whole heating/cooling cycle is fully automated, we hardly notice it being too hot / cold as it sorts itself out quickly.

At night you usually need to have a big zone open as well as a smaller bedroom zone if there's only a few zones open.

I'd tweak the zone dampers between 5-30% max.

**Comment by lmdc45** [comment]
Yes I agree usually left a central room as fan only also. However I think it would need control/update of fan opening as not too much is required compare to during the day.
Issue as well is that izone does not let me know when system send heat/cool/fan command. I know it is on heat mode, and when power consumption goes up and then supply temp I know it is heating, but not sure why there is no access to the actual heat command being sent or not that's annoying to monitor.

**Comment by Swamp-Ig** [comment]
I'm not sure I quite get what you're talking about. The integration allows you to change the setpoints for the zones when on zone control, plus allows you to monitor what mode it's set on.

**Comment by lmdc45** [comment]
When zone is on heat mode for instance it will wait for the temperature to drop below threshold before sending the actual heat command to the unit.
I don't see any way to check when that heat command is on or off, just can guess from energy consumption and supply temp starting to get higher that heat command is on.
